app/components/ReceiveTab.tsx

```diff
@@ -3,24 +3,28 @@ import { Button } from '@/components/ui/button'
 import { Card } from '@/components/ui/card'
 import { Label } from '@/components/ui/label'
 import { type FormEvent, useState } from 'react'
-import { receiveOffer } from '../lib/api'
+import { receiveOffer, receiveRequest } from '../lib/api'
 
 export function ReceiveTab() {
-  const [receiveCredentialOfferUri, setReceiveCredentialOfferUri] = useState<string>()
+  const [receiveCredentialOfferOrPresentationUri, setReceiveCredentialOfferUri] = useState<string>()
   const [receivedCredentials, setReceivedCredentials] = useState()
+  const [receivedPresentation, setReceivedPresentation] = useState()
 
   async function onSubmitReceiveOffer(e: FormEvent) {
     e.preventDefault()
-    if (!receiveCredentialOfferUri) return
-
-    setReceivedCredentials(await receiveOffer(receiveCredentialOfferUri))
+    if (!receiveCredentialOfferOrPresentationUri) return
+    if (receiveCredentialOfferOrPresentationUri.startsWith('openid-credential-offer://')) {
```

Comment:
I would not check based on the scheme, as we for paradym use https://paradym.id so we can use deep linking. And e.g. HAIP uses `As a way to invoke the Wallet, at least a custom URL scheme haip:// MUST be supported`.

So i think we should check whether `credential_offer` or `credential_offer_uri` params are present and extract those. That is the most robust method I think

Reply:
Can't open an issue for this as we're in a fork